dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/dubboserver/DubboServer.java

```diff
@@ -0,0 +1,1277 @@
+import static org.apache.dubbo.rpc.Constants.PROXY_KEY;
+import static org.apache.dubbo.rpc.Constants.SCOPE_KEY;
+import static org.apache.dubbo.rpc.Constants.SCOPE_LOCAL;
+import static org.apache.dubbo.rpc.Constants.SCOPE_REMOTE;
+import static org.apache.dubbo.rpc.Constants.TOKEN_KEY;
+import static org.apache.dubbo.rpc.cluster.Constants.EXPORT_KEY;
+import static org.apache.dubbo.rpc.cluster.Constants.REFER_KEY;
+
+public class DubboServer extends GenericEventListener {
+
+    private static final String NAME = DubboServer.class.getSimpleName();
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    private static volatile DubboServer instance;
+
+    private final DubboBootstrap dubboBootstrap;
+
+    private final AtomicBoolean awaited = new AtomicBoolean(false);
+
+    private final Lock lock = new ReentrantLock();
+
+    private final Condition condition = lock.newCondition();
+
+    private final Lock destroyLock = new ReentrantLock();
+
+    private final ExecutorService executorService = newSingleThreadExecutor();
+
+    private final EventDispatcher eventDispatcher = EventDispatcher.getDefaultExtension();
+
+    private final ExecutorRepository executorRepository = getExtensionLoader(ExecutorRepository.class).getDefaultExtension();
+
+    private final ConfigManager configManager;
+
+    private ReferenceConfigCache cache;
+
+    private volatile boolean exportAsync;
+
+    private volatile boolean referAsync;
+
+    private AtomicBoolean initialized = new AtomicBoolean(false);
+
+    private AtomicBoolean started = new AtomicBoolean(false);
+
+    private AtomicBoolean startup = new AtomicBoolean(true);
+
+    private AtomicBoolean destroyed = new AtomicBoolean(false);
+
+    private AtomicBoolean shutdown = new AtomicBoolean(false);
+
+    private volatile ServiceInstance serviceInstance;
+
+    private volatile MetadataService metadataService;
+
+    private volatile MetadataServiceExporter metadataServiceExporter;
+
+    private List<ServiceConfigBase<?>> exportedServices = new ArrayList<>();
+
+    private List<Future<?>> asyncExportingFutures = new ArrayList<>();
+
+    private List<CompletableFuture<Object>> asyncReferringFutures = new ArrayList<>();
+
+    private Map<ServiceConfigBase, Future<?>> serviceConfigBase2AsyncExportingFutures = new ConcurrentHashMap<>();
+
+    private Map<ReferenceConfigBase, CompletableFuture<Object>> referenceConfigBase2AsyncExportingFutures = new ConcurrentHashMap<>();
+
+    /**
+     * A delayed exposure service timer
+     */
+    private static final ScheduledExecutorService DELAY_EXPORT_EXECUTOR = Executors.newSingleThreadScheduledExecutor(new NamedThreadFactory("DubboServiceDelayExporter", true));
+
+    /**
+     * See {@link ApplicationModel} and {@link ExtensionLoader} for why DubboServer is designed to be singleton.
+     */
+    public static DubboServer getInstance(DubboBootstrap dubboBootstrap) {
+        if (instance == null) {
```

Comment:
Does it make sense to keep the dubbo server instance? The getInstance method with parameters looks strange, and the `DubboBootstrap` parameter is not used when the getInstance method is called for the second time.